server/src/main/java/io/spine/server/bus/BusFilter.java

```diff
@@ -40,18 +47,98 @@
      *
      * <p>A filter can:
      * <ul>
-     *     <li>accept the message (by returning {@code Optional.empty()};
+     *     <li>accept the message (by returning {@code Optional.empty()});
      *     <li>reject the message with {@link io.spine.base.Error Error} status, for example, if
      *         it fails to pass the validation;
      *     <li>reject the message with {@code OK} status. For example, a scheduled command may not
-     *         pass a filter.
+     *         pass a filter;
+     *     <li>reject the message with a business rejection. For example, a command can be rejected
+     *         if the user who made the request does not have enough permissions in the system.
+     *         Such rejection is only applicable to commands.
      * </ul>
      *
-     * @param envelope the envelope with the message to filter
+     * @param envelope
+     *         the envelope with the message to filter
      * @return {@code Optional.empty()} if the message passes the filter,
      *         {@linkplain Ack posting result} with either status otherwise
      */
-    Optional<Ack> accept(E envelope);
+    Optional<Ack> doFilter(E envelope);
```

Comment:
Not sure why rename that.

The idea of having a method like `doFilter()` is to avoid the clash with `filter()` in the public API. We don't have `filter()` here.